### PR TITLE
✨ Emit named method params = better signature help

### PR DESCRIPTION
**Pull request by roblav96**
```ts
public constructor(context: android.content.Context, serviceComponent: android.content.ComponentName, callback: android.support.v4.media.MediaBrowserCompat.ConnectionCallback, rootHints: android.os.Bundle);
```
instead of
```ts
public constructor(param0: android.content.Context, param1: android.content.ComponentName, param2: android.support.v4.media.MediaBrowserCompat.ConnectionCallback, param3: android.os.Bundle);
```

I've tested all tasks in the `Makefile` with no issues.

### CODE REVIEW

File: dts-generator/src/main/java/com/telerik/dts/DtsApi.java [hunk 1]
```diff
@@ -888,15 +889,43 @@ private String getMethodName(Method m) {
     }
 
     private String getMethodParamSignature(JavaClass clazz, TypeDefinition typeDefinition, Method m) {
+        Pattern pattern = Pattern.compile("\\s" + Pattern.quote(m.getName()) + "\\((.*)\\)\\s");
+        Matcher matcher = pattern.matcher(m.toString() + " ");
+        List<String> params = Arrays.asList();
+        if (matcher.find()) {
+            params = Arrays.asList(matcher.group(1).split("\\,\\s"));
+            if (matcher.find()) {
+                System.out.println("🆘 getMethodParamSignature: Second matcher.find() -> " + m.toString());
```
Comment (NathanaelA) [inline dts-generator/src/main/java/com/telerik/dts/DtsApi.java]: Seems like you have a weird extra "SOS" Unicode character here at the front of the println.

**Comment by NathanaelA** [comment]
@roblav96 - Looks good, can you sign the CLA.

File: dts-generator/src/main/java/com/telerik/dts/DtsApi.java [hunk 2]
```diff
@@ -888,15 +889,43 @@ private String getMethodName(Method m) {
...
+            }
+        } else {
+            System.out.println("🆘 getMethodParamSignature: !matcher.find() -> " + m.toString());
```
Comment (NathanaelA) [inline dts-generator/src/main/java/com/telerik/dts/DtsApi.java]: And here.

**Review by NathanaelA — CHANGES_REQUESTED** [review]

**Comment by roblav96** [comment]
@NathanaelA This commit [`54d5ff5` (#54)](https://github.com/NativeScript/android-dts-generator/pull/54/commits/54d5ff5be67c41cdb5094769288db0faf8711cd8) resolves a few issues, the primary concern is that on a rare occasion duplicate method definitions are produced yielding the same params type signature but with different names, which is most likely why `params0` was originally implemented.

What I've basically done is created a
```java
private Map<String, List<String>> realParams = new HashMap<>();
```
which stores method params based on it's signature, so that `abstract` or `generic` versions of the same method signature can use the params from the original `real` method param name's instead of `arg1` `arg2` `arg3`...

Here's some output examples:
- [android-platform-29.d.ts](https://gist.github.com/roblav96/36797b428e33e3c1f07b3081ce7ee8c6)
- [androidx-29.d.ts](https://gist.github.com/roblav96/548a2e5cb10ec13f4d3534585ee09843)
- [com.github.pedroSG94.rtmp-rtsp-stream-client-java_rtplibrary.d.ts](https://gist.github.com/roblav96/c57b2bcfaed3238afd37f6d03d24f226)

**Comment by NathanaelA** [comment]
When I get a chance I'm going to test this.   Looks like a AWESOME PR; thanks!

**Comment by NathanaelA** [comment]
Appears to be a bug somewhere in the new code;

Current DTS Generator:
```
declare module com {
        export module google {
                export module android {
                        export module material {
                                export module transformation {
                                        export class FabTransformationSheetBehavior extends com.google.android.material.transformation.FabTransformationBehavior {
                                                public static class: java.lang.Class<com.google.android.material.transformation.FabTransformationSheetBehavior>;
                                                public constructor();
                                                public onCreateMotionSpec(param0: globalAndroid.content.Context, param1: boolean): com.google.android.material.transformation.FabTransformationBehavior.FabTransformationSpec;
                                                public onExpandedStateChange(param0: globalAndroid.view.View, param1: globalAndroid.view.View, param2: boolean, param3: boolean): boolean;
                                                public constructor(param0: globalAndroid.content.Context, param1: globalAndroid.util.AttributeSet);
                                        }
                                }
                        }
                }
        }
}
```

With your patch:
```
declare module com {
        export module google {
                export module android {
                        export module material {
                                export module transformation {
                                        export class FabTransformationSheetBehavior extends com.google.android.material.transformation.FabTransformationBehavior {
                                                public static class: java.lang.Class<com.google.android.material.transformation.FabTransformationSheetBehavior>;
                                                public onCreateMotionSpec(context0: globalAndroid.content.Context, boolean1: boolean): com.google.android.material.transformation.FabTransformationBehavior.FabTransformationSpec;
                                                public constructor();
                                                public onCreateMotionSpec(context: globalAndroid.content.Context, expanded: boolean): com.google.android.material.transformation.FabTransformationBehavior.FabTransformationSpec;
                                                public constructor(context: globalAndroid.content.Context, attrs: globalAndroid.util.AttributeSet);
                                                public onExpandedStateChange(dependency: globalAndroid.view.View, child: globalAndroid.view.View, expanded: boolean, animated: boolean): boolean;
                                        }
                                }
                        }
                }
        }
}
```
Notice it has two `onCreateMotionSpec` with almost the same specification but the second one is using number postfix's.
In addition the file created has 29834 lines vs 29449 lines for the original DTS generator, so it added another 400ish probably duplicate lines like this one that I am pointing out.

Please note; the comparison is with the current master; and using the cool new test feature `make test-compare-output` that was merged with master -- it creates a test based on the output of all the jars in the lib folder.   Of course with your DTS changes the test will fail (expected); but the number of lines output should match and the classes shouldn't have extra duplicate functions...

**Pull request closed without merging**